src/server/streamable-http.ts

```diff
@@ -0,0 +1,496 @@
+    } else {
+      res.writeHead(405, this._customHeaders).end(JSON.stringify({
+        jsonrpc: "2.0",
+        error: {
+          code: -32000,
+          message: "Method not allowed"
+        },
+        id: null
+      }));
+    }
+  }
+
+  /**
+   * Handles GET requests to establish SSE connections
+   */
+  private async handleGetRequest(req: IncomingMessage, res: ServerResponse): Promise<void> {
+    // validate the Accept header
+    const acceptHeader = req.headers.accept;
+    if (!acceptHeader || !acceptHeader.includes("text/event-stream")) {
+      res.writeHead(406, this._customHeaders).end(JSON.stringify({
+        jsonrpc: "2.0",
+        error: {
+          code: -32000,
+          message: "Not Acceptable: Client must accept text/event-stream"
+        },
+        id: null
+      }));
+      return;
+    }
+
+    const connectionId = randomUUID();
+    const lastEventId = req.headers["last-event-id"];
+    const lastEventIdStr = Array.isArray(lastEventId) ? lastEventId[0] : lastEventId;
+
+    // Prepare response headers
+    const headers: Record<string, string> = {
+      ...this._customHeaders,
+      "Content-Type": "text/event-stream",
+      "Cache-Control": "no-cache",
+      Connection: "keep-alive",
+    };
+    
+    // Only include session ID header if session management is enabled
+    if (this._enableSessionManagement) {
+      headers["mcp-session-id"] = this._sessionId;
+    }
+
+    res.writeHead(200, headers);
+
+    const connection: StreamConnection = {
+      response: res,
+      lastEventId: lastEventIdStr,
+      messages: [],
+    };
+
+    this._connections.set(connectionId, connection);
+
+    // if there is a Last-Event-ID, replay messages on this connection
+    if (lastEventIdStr) {
+      this.replayMessages(connectionId, lastEventIdStr);
+    }
+
+    res.on("close", () => {
+      this._connections.delete(connectionId);
+      // remove all request mappings associated with this connection
+      for (const [reqId, connId] of this._requestConnections.entries()) {
+        if (connId === connectionId) {
+          this._requestConnections.delete(reqId);
+        }
+      }
+      if (this._connections.size === 0) {
+        this.onclose?.();
+      }
+    });
+  }
+
+  /**
+   * Handles POST requests containing JSON-RPC messages
+   */
+  private async handlePostRequest(req: IncomingMessage, res: ServerResponse, parsedBody?: unknown): Promise<void> {
+    try {
+      // validate the Accept header
+      const acceptHeader = req.headers.accept;
+      if (!acceptHeader || 
+         (!acceptHeader.includes("application/json") && !acceptHeader.includes("text/event-stream"))) {
+        res.writeHead(406).end(JSON.stringify({
+          jsonrpc: "2.0",
+          error: {
+            code: -32000,
+            message: "Not Acceptable: Client must accept application/json and/or text/event-stream"
+          },
+          id: null
+        }));
+        return;
+      }
```

Comment:
According to the spec:
>The client MUST include an Accept header, listing both application/json and text/event-stream as supported content types.

So this needs to be changed to `and` instead of `and/or`